fili-core/src/main/java/com/yahoo/bard/webservice/logging/RequestLogUtils.java

```diff
@@ -0,0 +1,352 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.logging;
+
+import com.yahoo.bard.webservice.config.SystemConfig;
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+import com.yahoo.bard.webservice.logging.RequestLog.TimedPhase;
+import com.yahoo.bard.webservice.logging.blocks.Durations;
+import com.yahoo.bard.webservice.logging.blocks.Threads;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static com.yahoo.bard.webservice.druid.client.impl.AsyncDruidWebServiceImpl.DRUID_QUERY_TIMER;
+import static com.yahoo.bard.webservice.logging.RequestLog.LOG_ID_KEY;
+import static com.yahoo.bard.webservice.util.StreamUtils.not;
+import static com.yahoo.bard.webservice.web.handlers.workflow.DruidWorkflow.REQUEST_WORKFLOW_TIMER;
+import static com.yahoo.bard.webservice.web.handlers.workflow.DruidWorkflow.RESPONSE_WORKFLOW_TIMER;
+
+/**
+ * An interface for application code to interact with the {@link RequestLog}.
+ */
+public class RequestLogUtils {
+    private static final String TOTAL_TIMER = "TotalTime";
+    private static final Logger LOG = LoggerFactory.getLogger(RequestLog.class);
+    private static final SystemConfig SYSTEM_CONFIG = SystemConfigProvider.getInstance();
+    private static final List<String> LOGINFO_ORDER = generateLogInfoOrder();
+
+    private static final ThreadLocal<RequestLog> RLOG = ThreadLocal.withInitial(RequestLog::new);
+
```

Comment:
This is an interesting way to do this.  I wonder if SystemConfig should be tweaked in a similar way.